### PR TITLE
refactor: Safer command interpolation (DEV-3184)

### PR DESCRIPTION
**Pull request by siers**

### CODE REVIEW

**Review by BalduinLandolt — APPROVED** [review]
I'm no specialist, but I suppose some injections would still be possible, right? (If you have a file named `foo.jpeg ; echo boom ; echo gotcha.jpeg`?)

**Comment by siers** [comment]
You should only be able to pass arguments to the container, but they wouldn't be treated as a bash command unless the container would also try to interpret it that way.

So ";" is a literal string that would get passed to an argument.

File: src/main/scala/swiss/dasch/infrastructure/CommandExecutorLive.scala [hunk 1]
```diff
@@ -53,21 +53,29 @@ trait CommandExecutor {
...
         assetDir <- storageService.getAssetsBaseFolder().flatMap(_.toAbsolutePath).orDie
-      } yield Command(
-        s"docker run --entrypoint $command -v $assetDir:$assetDir daschswiss/knora-sipi:${BuildInfo.sipiVersion} $params"
-      )
+      } yield {
+        Command(
+          List(
+            List("docker", "run", "--entrypoint", command),
+            List("-v", s"$assetDir:$assetDir"),
+            List(s"daschswiss/knora-sipi:${BuildInfo.sipiVersion}"),
```
Comment (mpro7) [inline src/main/scala/swiss/dasch/infrastructure/CommandExecutorLive.scala:71]: Could these two interpolations also be replaced or it doesn't make sense? I'm not sure are these variables strings already or not, that's why added `toString` conversion to them - remove if not necessary.
```suggestion
            List("-v", "assetDir.toString, ":", "assetDir.toString"),
            List("daschswiss/knora-sipi:", "BuildInfo.sipiVersion.toString"),
```

**Review by mpro7 — APPROVED** [review]
Thanks! I have is one suggestion:

File: src/main/scala/swiss/dasch/infrastructure/CommandExecutorLive.scala (hunk 1 above)
Comment (siers) [reply src/main/scala/swiss/dasch/infrastructure/CommandExecutorLive.scala:71]: That would change the meaning – they have to be in the same argument, now they would be in two.

A very imperfect analogy would be `ls directory/file` vs `ls directory file`. The analogy here is that the space would split the arguments into two, which would change the meaning of code entirely.

Comment (mpro7) [reply src/main/scala/swiss/dasch/infrastructure/CommandExecutorLive.scala:71]: As far as it does not expose it to vulnerabilities mentioned in the task I'm fine with what you did :)

**Review by seakayone — APPROVED** [review]

**Pull request merged**